tradebranch

Derar1 opened the pull request:
trading branch, 85% chance of being able to trade with someone. 15% chance no one wants to trade. worms, food,money, are weighed differently from everything else. trading/no trading wastes a day

Commit:
tradebranch

ayebear commented on client/js/states/trading.js:
```diff
@@ -0,0 +1,81 @@
+		let item = randNonZeroKey(party.supplies);
+		let item1 = randNonZeroKey(party.supplies);
+		
+		while (item == item1){ // not trading the same thing 
```
Possible infinite loop, it'd be better to just remove the first one so the second pick won't ever pick the first.

ayebear committed:
Converted spaces to tabs

ayebear committed:
Improved trading
Fixed infinite loop issue if you have no supplies
Added randomized person names that you can trade with
Simplified description generation with a function, used by the river crossing and trading states so far
Added worms to "check supplies" state
Fixed and improved other things
